knowagetalendengine/src/main/java/it/eng/spagobi/engines/talend/runtime/JobDeploymentDescriptor.java

```diff
@@ -1,122 +1,134 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0, without the "Incompatible With Secondary Licenses" notice.
  * If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 package it.eng.spagobi.engines.talend.runtime;
 
-import it.eng.spagobi.engines.talend.TalendEngine;
-import it.eng.spagobi.engines.talend.TalendEngineConfig;
-
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.InputStream;
 
+import javax.xml.XMLConstants;
+
+import org.apache.log4j.Logger;
 import org.dom4j.Document;
 import org.dom4j.DocumentException;
 import org.dom4j.Node;
 import org.dom4j.io.SAXReader;
+import org.xml.sax.SAXException;
 
 /**
  * @author Andrea Gioia
  *
  */
 public class JobDeploymentDescriptor {
+
+	private static final Logger LOGGER = Logger.getLogger(JobDeploymentDescriptor.class);
+
 	String project;
 	String language;
-		
+
 	/**
 	 * Instantiates a new job deployment descriptor.
 	 */
-	public JobDeploymentDescriptor() {}
-	
-		
+	public JobDeploymentDescriptor() {
+	}
+
 	/**
 	 * Instantiates a new job deployment descriptor.
-	 * 
-	 * @param project the project
+	 *
+	 * @param project  the project
 	 * @param language the language
 	 */
 	public JobDeploymentDescriptor(String project, String language) {
 		this.project = project;
 		this.language = language;
-	}		
-	
+	}
+
 	/**
 	 * Load.
-	 * 
+	 *
 	 * @param file the file
-	 * 
+	 *
 	 * @throws FileNotFoundException the file not found exception
-	 * @throws DocumentException the document exception
+	 * @throws DocumentException     the document exception
 	 */
 	public void load(File file) throws FileNotFoundException, DocumentException {
 		load(new FileInputStream(file));
 	}
-	
+
 	/**
 	 * Load.
-	 * 
+	 *
 	 * @param is the is
-	 * 
+	 *
 	 * @throws DocumentException the document exception
 	 */
-	public void load(InputStream is) throws DocumentException{
+	public void load(InputStream is) throws DocumentException {
 		SAXReader reader = new org.dom4j.io.SAXReader();
-	    Document document = null;
-	    
-	   	document = reader.read(is);		
-	    
-	    Node job = document.selectSingleNode("//etl/job");
-	    if (job != null) {
-	    	this.project = job.valueOf("@project");
-	    	this.language = job.valueOf("@language");
-	    }
+
+		try {
+			reader.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+			reader.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+			reader.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+			reader.setFeature("http://xml.org/sax/features/external-general-entities", false);
+		} catch (SAXException | SecurityException e) {
+			LOGGER.error("Error loading XML document: " + e.getMessage(), e);
+			throw new SecurityException("Error loading XML document: " + e.getMessage(), e);
```

Comment:
The error message is misleading: we are not loading XML document in that block

Reply:
Fixed